models/user_input.js

```diff
@@ -0,0 +1,174 @@
+                        });
+                }
+            });
+        },
+
+        create: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => {
+                validateUserInputData(data)
+                    .then(function () {
+                        return _db.query(
+                            'INSERT INTO user_inputs ' +
+                            '(experiment_id, device_id, user_id,' +
+                            'description, timestamp, created_at, updated_at) ' +
+                            'VALUES ($1, $2, $3, $4, $5, $6, $6) returning experiment_id, device_id',
+                            [data.experiment_id, data.device_id, data.user_id,
+                            data.description, data.timestamp, time]);
+                    })
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        reject(err);
+                    });
+            });
+        },
+
+        delete: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => {
+                _db.query('UPDATE user_inputs SET deleted_at = $2 WHERE id = $1 returning id', [data.id, time])
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        reject(err);
+                    });
+            });
+        },
+
+        updateDescription: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => { 
+                _db.query('UPDATE user_inputs SET description = $2, updated_at = $3 WHERE id = $1 and deleted_at = 0 returning description',
+                    [data.id, data.description, time])
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        reject(err);
+                    });
+            });
+        },
+    }
+};
+
+function findOneById(id) {
+    return new Promise((resolve, reject) => {
+      _db.query('SELECT * FROM user_inputs WHERE id = $1 and deleted_at = 0', [id])
+        .then((result) => {
+          if (result.rows[0]) {
+            resolve(result.rows[0]);
+          }
+          else {
+            reject('no user input found')
+          }
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    });
+}
+
+function validateUserInputData(data) {
+    return new Promise((resolve, reject) => {
+        if (!data.description) {
+            reject('invalid input: description missing')
+        }
+        else{
+            columns = ['user_id','device_id', 'description', 'experiment_id']
```

Comment:
We should probably only allow admins to set the `user_id` to that of a different user. In the controller hydrateReq, this could be added.

Reply:
👍

Reply:
Actually could you please explain what you mean by the last bit of your statement?

Reply:
In the user inputs controller, you had a hydrateReq function. You can use that function to make sure that the user_id is set to the own users id unless he's an admin. Similar to how it's done in the users controller

Reply:
Okay. That was my initial understanding but you also asked me to take out hydratedReq https://github.com/Jss7268/KGCOESeniorProjectAPI/pull/19#discussion_r289535371 That was why I got confused. To confirm, you take back your comment to take out `hydrateReq`

Reply:
The hydrateReq that you had initially wasn't doing anything so it wasn't needed. However, if you want to make it so that the user_id is the uid of the current user, then you can put that logic into a function such as the hydrateReq function in user.js

Reply:
👍

Reply:
I think I'd like to handle this in a separate PR